Sources/Sentry/include/SentryDefaultRateLimits.h

```diff
@@ -1,15 +1,24 @@
 #import <Foundation/Foundation.h>
 #import "SentryRateLimits.h"
 
+@class SentryRetryAfterHeaderParser;
+@class SentryRateLimitParser;
+
 NS_ASSUME_NONNULL_BEGIN
 
 /**
- * Implements only rate limits with status code 429. Needs to be updated.
+ Parses HTTP responses from the Sentry server for rate limits and stores them
+ in memory. The server can communicate a rate limit either through the 429 status
+ code with a "Retry-After" header or through any response with a custom
+ "X-Sentry-Rate-Limits" header. This class is thread safe.
 */
 NS_SWIFT_NAME(DefaultRateLimits)
 @interface SentryDefaultRateLimits : NSObject <SentryRateLimits>
 
-- (BOOL)isRateLimitReached;
+- (instancetype) initWithParsers:(SentryRetryAfterHeaderParser *)retryAfterHeaderParser
+                 rateLimitParser:(SentryRateLimitParser *)rateLimitParser;
```

Comment:
Even though much uglier, shouldn't the name actually be something like: 

```suggestion
- (instancetype) initWithRetryAfterHeaderParser:(SentryRetryAfterHeaderParser *)retryAfterHeaderParser
                 rateLimitParser:(SentryRateLimitParser *)rateLimitParser;
```
Where it says the first thing it's instantiated with, in this case `SentryRetryAfterHeaderParser`?

Reply:
Good point, that's the way to do it in Objective-C. I'm still not 100% used to its verbosity.